xnames/generator/types.go.tpl

```diff
@@ -36,6 +36,11 @@ type {{ $xnameType.Entry.Type }} struct {
 {{- end }}
 }
 
+// Type will return the corresponding HMSType
```

Comment:
can you tell me more about .go.tpl files? im not familiar with them.

Reply:
Here is some more context: https://github.com/Cray-HPE/hms-xname#code-generation

The .go.tpl files are template files that the generator uses to generate the real types.go and util.go. It uses the golang template language: https://pkg.go.dev/text/template